### DISCUSSION TITLE
A way to set up Keka as a default app to manage archive files for all the users of a Mac?

### DISCUSSION THREAD

**Discussion by ibob-z** [aonez/Keka#1208] [Q&A] [answered]
Hi, I am an IT engineer at an International school.

Just curious if there's a way to set up Keka as a  default app to manage archive files for all the users of a Mac.

A script may easier to deploy Keka remotely.

Our staff is using their Active Directory account to log in to their working Mac, so if I could use a script to set up Keka for their mobile account is easier than letting them set up via a manual.

if you would be able to assist me in any way I would greatly appreciate that. Thanks a lot.

**Comment by aonez** [comment]
Hi @ibob-z,

You can run the `KekaExternalHelper.app` (https://www.keka.io/defaultapp) once (or every time, up to you). This helper sets Keka as the default extraction tool for all the supported formats. You have all info of this helper here: https://github.com/aonez/Keka/wiki/Default-application

I could slightly modify this helper so it runs silently (no confirmation dialog).

Let me know if this will suit your need or you may need a modification/alternative.

**Comment by aonez** [comment, marked as answer]
Went ahead and update the helper (v1.1.3) to be executed from anywhere with no GUI, using the argument `-s` or `--set-as-default`.

So executing it like `KekaExternalHelper.app/Contents/MacOS/KekaExternalHelper --set-as-default` will find the installed Keka and set as the default app for all it's supported extraction formats on the user that executed the command.

You can place the helper anywhere you want, just be sure it has no quarantine applied.

**Reply by ibob-z** [reply]
Hi @aonez, thanks so much for your quick response, and sorry I didn't dig into the Helper's instruction page before I ask.
Here's some feedback.
I've tried running the script via the "sudo" command and no matter whether I place the KekaExternalHelper.app in the Desktop or the Applications folder, it showed "Keka is now set as the default app", but under both my current user or a new user, the macOS still use the Archive Utility.app to unzip a .zip file. When I go to the Preferences panel and try to click "Set Keka as the default uncompressor", it shows "An extension is required" to guide me to download the KekaDefaultApp v1.1.3, **so far, the only way to set Keka as a default uncompressor for current user is run the KekaExternalHelper.app from the Applications folder manually then went to the Preferences panel and click "Set Keka as the default uncompressor"**.

My usage scenario is: different employees will use their own network accounts to log in to the same computer. I want to make sure that after they log in to their accounts, they can directly use Keka as the default uncompressor software.

But now I don't know how to make Keka the default uncompressor software for any user.

I have tested this on macOS Monterey 12.6.3.

**Reply by aonez** [reply]
No need to use `sudo` at all, just don't do it 😛

I'll do some tests and get back to you.

**Reply by ibob-z** [reply]
Hi @aonez, I have written the below .plist file and placed it in the /Library/LaunchAgents/ folder, the .plist file has Read/Write/Execute rights for all users/groups, my thoughts is if a new user is the first time to log in to this computer, the startup .plist will check if there's an invisible file(_~/.first_login_complete_) under the current user's home folder, if it not exists, then run the command "/Applications/KekaExternalHelper.app/Contents/MacOS/KekaExternalHelper --set-as-default", then create an invisible file(.first_login_complete) under the current user's home folder for the next time checking. However, I failed and don't know the reason that the .plist startup not be executed under macOS 13.3.1(I haven't had a chance to test it under a different macOS). The Gatekeeper is disabled on the test computer. Any suggestion/idea that can help me to execute the .plist startup item? Many thanks.

```
<?xml version="1.0" encoding="UTF-8"?>

<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>Label</key>
    <string>com.bcis.KekaDefaultApp</string>
    <key>ProgramArguments</key>
    <array>
        <string>/usr/bin/env</string>
        <string>bash</string>
        <string>-c</string>
        <string>if [ ! -f ~/.first_login_complete ]; then /Applications/KekaExternalHelper.app; KekaExternalHelper.app/Contents/MacOS/KekaExternalHelper --set-as-default; touch ~/.first_login_complete; fi</string>
    </array>
    <key>RunAtLoad</key>
    <true/>
</dict>
</plist>
```

**Reply by ibob-z** [reply]
[https://developer.apple.com/forums/thread/718907](url)
I will try it on a lower version of macOS and will get back to you.

**Reply by ibob-z** [reply]
Hi @aonez, I have tried under macOS 12.6.5 and 13.3.1(a) too, the .plist startup didn't be executed...
I have also tried to use sudo launchctl load to launch it manually, but failed too, it says
`Path had bad ownership/permissions`
`Load failed:  122:  Path had bad ownership/permissions`
The .plist file I have tried to give it _current user_(for example user A) as the owner, _admin_ as the owner group, or _root_ as the owner and _wheel_ as the owner group. The .plist file's current permission like the below,
`-rwxrwxrwx  1 a   admin`
and I have removed all the quarantine attributes for the .plist file. The Gatekeeper and SIP are disabled on the test computer.
I placed the KekaExternalHelper(v.1.1.3) and Keka(v1.2.61) under the Applications folder and removed all quarantine attributes too.
I don't have any coding experience, any suggestions?

**Reply by ibob-z** [reply]
Solved. Never mind. I wish you good health and develop more apps as good as Keka.

**Reply by aonez** [reply]
Sorry @ibob-z I was very busy. It works now for you then?

**Reply by ibob-z** [reply]
No worries. I have written a .sh script file and used a new .plist file to load the shell script when a new user login. The wrong .plist file permissions(I set the permission to 777, but yesterday I found it should be 644) caused the above issue. I tested it under macOS 13.3.1 and it worked for both admin users and standard users.

**Reply by ibob-z** [reply]
Dear @aonez , it's me again. I'm facing an issue which is on macOS Tahoe 26.x, there will be more than ten windows that pop up asking the user to confirm whether they want to use Keka to open various types of compressed files. I'm not sure if you have a solution for this, but if not, are there more parameters(not just --set-as-default) available when I execute KekaExternalHelper.app/Contents/MacOS/KekaExternalHelper --set-as-default? For example, the parameter allows us to set only .zip&.rar&.7z files to open with Keka by default, so Mac user no need to click too many times to confirm.
